packages/core/src/types.ts

```diff
@@ -38,12 +38,12 @@ export interface ServerlessConfig extends Config {
 }
 
 export interface BeforeNotifyHandler {
-  (notice?: Notice): boolean | void
+  (notice?: Notice): boolean | void | Promise<boolean | void>
 }
 
 export interface AfterNotifyHandler {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  (error: any, notice?: Notice): boolean | void
+  (error: any, notice?: Notice): boolean | void | Promise<void>
```

Comment:
Can the `AfterNotifyHandler` resolve to a boolean? Just `void`?

Reply:
Hmm, I missed the fact that we allow a boolean to be returned from this handler. I looked at the code and we are not really looking at the result of these handlers, so I think we _could_ remove the `boolean` return option as well.

Reply:
Done!